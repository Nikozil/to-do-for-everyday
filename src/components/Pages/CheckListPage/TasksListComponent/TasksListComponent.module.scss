@import '../../../../App.scss';

.taskList__comment {
  color: $comment-color;
  font-size: 0.8em;
}
