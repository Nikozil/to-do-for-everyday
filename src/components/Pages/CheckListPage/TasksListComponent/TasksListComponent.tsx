import { endOfToday, getTime } from 'date-fns';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  checkTask,
  deleteTask,
  LivedTask,
  PartialTaskData,
  Task,
  uncheckTask,
  updateTask,
} from '../../../../Redux/modules/tasksSlice';
import { AppStateType } from '../../../../Redux/store';
import DoneTaskComponent from './DoneTaskComponent/DoneTaskComponent';
import TaskComponent from './TaskComponent/TaskComponent';
import styles from './TasksListComponent.module.scss';

const TasksListComponent = () => {
  const tasks = useSelector((state: AppStateType) => state.tasks.tasksList);
  const undoneTasks = tasks.filter(
    (i) => !i.data.done && i.data.time <= getTime(endOfToday())
  );
  const doneTasks = useSelector(
    (state: AppStateType) => state.tasks.livedDay.doneTasksList
  );

  const dispatch = useDispatch();

  const taskComponentCheckHandler = (task: Task) => {
    dispatch(checkTask(task));
  };
  const taskComponentUncheckHandler = (task: LivedTask) => {
    dispatch(uncheckTask(task));
  };
  const taskComponentRepeatHandler = (id: string, data: PartialTaskData) => {
    dispatch(updateTask(id, data));
  };
  const taskComponentDeleteHandler = (id: string) => {
    dispatch(deleteTask(id));
  };

  const mapTaskComponent = (task: Task) => (
    <TaskComponent
      task={task}
      key={task.id}
      checkHandler={taskComponentCheckHandler}
      repeatHandler={taskComponentRepeatHandler}
      deleteHandler={taskComponentDeleteHandler}
    />
  );
  const mapDoneTaskComponent = (task: LivedTask) => (
    <DoneTaskComponent
      task={task}
      key={task.id}
      uncheckHandler={taskComponentUncheckHandler}
    />
  );

  return (
    <>
      <span>Задачи</span>
      <ul className={styles.list}>
        {undoneTasks.length ? (
          undoneTasks.map(mapTaskComponent)
        ) : (
          <span className={styles['taskList__comment']}>Нет новых задач</span>
        )}
      </ul>

      <span>Выполнено</span>
      <ul className={styles.list}>
        {doneTasks.length ? (
          doneTasks.map(mapDoneTaskComponent)
        ) : (
          <span className={styles['taskList__comment']}>
            Задачи не выполнены
          </span>
        )}
      </ul>
    </>
  );
};

export default TasksListComponent;
