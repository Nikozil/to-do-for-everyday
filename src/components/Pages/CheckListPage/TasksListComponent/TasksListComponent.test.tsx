import React from 'react';
import { render } from '../../../../utils/tests/test-utils';
import TasksListComponent from './TasksListComponent';
import {
  initialState,
  LivedTask,
  Task,
} from '../../../../Redux/modules/tasksSlice';

const testTask = {
  id: '123',
  data: { name: 'task0', repeat: 0, done: false, time: new Date().getTime() },
} as Task;
const testDoneTask = { id: '123', name: 'task0' } as LivedTask;
describe('TasksListComponent', () => {
  it('render correct with tasks', async () => {
    const { asFragment } = render(<TasksListComponent />, {
      preloadedState: {
        tasks: {
          ...initialState,
          tasksList: [testTask],
          livedDay: { score: 3, tag: 'OHOHO', doneTasksList: [testDoneTask] },
        },
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });
  it('render correct without tasks', async () => {
    const { asFragment } = render(<TasksListComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
