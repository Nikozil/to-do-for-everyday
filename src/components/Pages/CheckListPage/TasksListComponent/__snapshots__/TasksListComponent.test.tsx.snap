// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksListComponent render correct with tasks 1`] = `
<DocumentFragment>
  <h2
    class="mb-3"
  >
    Задачи
  </h2>
  <ul
    class="list"
  >
    <li
      class="content__li"
    >
      <div
        class="content content_border"
      >
        <span
          class="task"
        >
          <span
            class="task__taskName"
          >
            task0
          </span>
        </span>
        <span
          class="task__buttons"
        >
          <button
            aria-label="options"
            class="button"
            data-testid="options"
            name="options"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 7h0.25c0.412 0 0.75-0.338 0.75-0.75v-2.5c0-0.413-0.338-0.75-0.75-0.75h-0.25v-3h-2v3h-0.25c-0.412 0-0.75 0.337-0.75 0.75v2.5c0 0.412 0.338 0.75 0.75 0.75h0.25v9h2v-9zM12 4h2v2h-2v-2zM9.25 13c0.412 0 0.75-0.338 0.75-0.75v-2.5c0-0.412-0.338-0.75-0.75-0.75h-0.25v-9h-2v9h-0.25c-0.412 0-0.75 0.338-0.75 0.75v2.5c0 0.412 0.338 0.75 0.75 0.75h0.25v3h2v-3h0.25zM7 10h2v2h-2v-2zM4.25 7c0.412 0 0.75-0.338 0.75-0.75v-2.5c0-0.413-0.338-0.75-0.75-0.75h-0.25v-3h-2v3h-0.25c-0.413 0-0.75 0.337-0.75 0.75v2.5c0 0.412 0.337 0.75 0.75 0.75h0.25v9h2v-9h0.25zM2 4h2v2h-2v-2z"
              />
            </svg>
             
          </button>
          <button
            aria-label="check-button"
            class="button task__buttons_left-margin"
            data-testid="check-button"
            name="check-button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM14 14h-12v-12h12v12z"
              />
            </svg>
             
          </button>
        </span>
      </div>
    </li>
  </ul>
  <h2
    class="mb-3"
  >
    Выполнено
  </h2>
  <ul
    class="list"
  >
    <li
      class="content__li"
    >
      <div
        class="content content_border"
      >
        <span
          class="task"
        >
          <span
            class="task__doneTaskName"
          >
            task0
          </span>
        </span>
        <span
          class="task__buttons"
        >
          <button
            aria-label="uncheck-button"
            class="button"
            data-testid="uncheck-button"
            name="uncheck-button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM7 12.414l-3.707-3.707 1.414-1.414 2.293 2.293 4.793-4.793 1.414 1.414-6.207 6.207z"
              />
            </svg>
             
          </button>
        </span>
      </div>
    </li>
  </ul>
   
</DocumentFragment>
`;

exports[`TasksListComponent render correct without tasks 1`] = `
<DocumentFragment>
  <h2
    class="mb-3"
  >
    Задачи
  </h2>
  <span
    class="stub"
  >
    Нет новых задач
  </span>
  <h2
    class="mb-3"
  >
    Выполнено
  </h2>
  <span
    class="stub"
  >
    Список пуст
  </span>
   
</DocumentFragment>
`;
